verus/lcf/README.md

```diff
@@ -0,0 +1,396 @@
+
+## Verus specific features in implementation of kernel                  
+
+One Verus specific feature in the implementation of the kernel is the separation between spec and exec code. Exec code is executable code such as functions susbt, mk_lf, and mk_thm that are commonly used in the kernel and are compileed. Spec code is code that reasons about the exec code without compiling and returns errors to the verifier if the code is not correct. Another is the use of loop invariants, triggers, assertions, and lemmas that were used to aid the verifier in reasoning about preconditions, postconditions, and recommendations. The ones left are essential for kernel verification, so do not remove them. They have been thoroughly commented and described in the code body.
+
+Another verus specific feature used was deep_view which allowed us to access the deep_view of exec functions and call spec functions in exec mode when writing invariants, preconditions, poscondtions, and asserts. Additional ensures statements were used to help the verifier understand that the deep_view call could be before or after accessing fields of dat types and also reasoning that the exec versions and spec versions functioned the same way. here's an example:
+
+    exec version:
+
+    pub fn wf(self) -> (res: bool)
+        ensures
+            res <==> self.deep_view().spec_wf(),
+    {
+      //using flag to check if all elements in the body are well formed
+        let mut flag = true;
+        assert(forall|k: int|
+            0 <= k < self.rs.len() ==> (#[trigger] self.rs[k].deep_view())
+                == self.deep_view().rs[k]);
+        for i in 0..self.rs.len()
+            invariant
+                0 <= i <= self.rs.len(),
+                //invariant forall checks all body elements if they are well formed using deep_view
+                flag <==> forall|j: int| #![auto] 0 <= j < i ==> self.rs[j].deep_view().spec_wf(),
+        {
+            flag = self.rs[i].wf() && flag
+        }
+        flag
+    }
+
+    spec version:
+
+    pub open spec fn spec_wf(self) -> bool {
+        forall|i: int| #![auto] 0 <= i < self.rs.len() ==> self.rs[i].spec_wf()
+    }
+
+
+
+
+TmpStringHashMap was another Verus specific feature in the implementation of the kernel. Developed by Jay Lorch, the StringHashMap data structure was introduced to the vstd library for Verus. Our team utilized addtional functions such as deep_view and partial_eq for StringHashMap, so the attached TmpStringHashMap file in the repository is the version that includes that functions that are used.
+
+## Missing features
+
+Some missing features from this verified kernel in Verus versus the original devleoped in Dafny are:
+1. BuiltInOp support
+2. #[verifier::external_body] used in some palces due to limitations in Verus (needs to be changed when support is added in Verus)
+
+## Toy Example of kernel working
```

Comment:
I don't think this section is required in this document---instead, you should just include the Datalog code you have in an inline comment in the Verus file.